src/prometheus_alert_rules/prometheus_tsdb_wal_truncations_failed.rule

```diff
@@ -0,0 +1,9 @@
+# Based on https://awesome-prometheus-alerts.grep.to/rules.html#prometheus-self-monitoring-1
+alert: PrometheusTsdbWalTruncationsFailed
+expr: increase(prometheus_tsdb_wal_truncations_failed_total{%%juju_topology%%}[1m]) > 0
+for: 0m
+labels:
+  severity: critical
+annotations:
+  summary: Prometheus TSDB WAL truncations failed (instance {{ $labels.instance }})
+  description: "Prometheus encountered {{ $value }} TSDB WAL truncation failures\n  VALUE = {{ $value }}\n  LABELS = {{ $labels }}"
```

Comment:
```suggestion
  description: |
    Prometheus encountered {{ $value }} TSDB WAL truncation failures.
    VALUE = {{ $value }}
    LABELS = {{ $labels }}
```

Reply:
Have a look at https://github.com/canonical/prometheus-k8s-operator/pull/199/files which contains prior art for this. Would be good to include Grafana Dashboard, for instance, in this PR as well.